### PR TITLE
change tree node property to a method

### PR DESCRIPTION
**Pull request by stevenbird**
changed tree node property to a method, and added a set_node method
(following the model of prob/set_prob etc in probability.py)

### CODE REVIEW

**Comment by stevenbird** [comment]
This is to partially address https://github.com/nltk/nltk/issues/475

**Comment by kmike** [comment]
I''m -0 on this because changing 'node' from variable to function is backwards-incompatible. That's why I wrote 'get_node' - we can keep backwards compatibility that way. 

As you mentioned, 'get_node' is not consistent with nltk.probability where 'get_' prefix is not used. 

What do you think about creating a function with some other name, e.g. label(), and keeping 'node' attribute? We could make this attribute property and issue a warning on first read or write access to this attribute.

I'm not an expert, but "node" definition in NLTK Tree looks unusual. Usually node is a data structure that contains some label (named either key or value, depending on application) and a pointer to children (in Python it would be a list of children). So what is called 'node' here is usually called 'node label'. Some references:
- In [Data Structures and Algorithms by Aho, Ullman and Hopcroft](http://www.amazon.com/Data-Structures-Algorithms-Alfred-Aho/dp/0201000237) they distinguish between 'node' and 'node label': nodes can be named, but these names must be unique, while 'node label' or 'node value' is some data associated with the node.
- In  Knuth vol 1 chapter 2 I haven't found this distinction explicit; node labels are called 'nodes' when this is not ambiguous. But when this distinction is important Knuth uses INFO(P) to denote information associated with node pointer P.
- [Wikipedia](http://en.wikipedia.org/wiki/Tree_%28data_structure%29) says that 'each node is a data structure consisting of a value, together with a list of references to nodes (the "children")'.

**Comment by heatherleaf** [comment]
+1 to @kmike's suggestion: `label()` and `set_label()`, and deprecate the `node` property

[edit: of course it should be `set_label` and not `setlabel`, since it is `set_prob`]

**Comment by stevenbird** [comment]
@kmike that's a good point -- the node includes the label, and they are not the same thing. I'll update the pull request (though its going to be a while now as I'm leaving for IJCNLP now.)

**Comment by stevenbird** [comment]
New changes support label() method as proposed by @kmike

File: nltk/toolbox.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def _record_parse(self, key=None, **kwargs):
     def _tree2etree(self, parent):
         from nltk.tree import Tree
 
-        root = Element(parent.node)
+        root = Element(parent._node)
```
Comment (kmike) [inline nltk/toolbox.py]: Is it correct?

Comment (stevenbird) [inline nltk/toolbox.py]: I missed that one. I'll fix it and do some more searching.

**Comment by stevenbird** [comment]
In the absence of any controversy I'm merging this now.

**Pull request merged**